### PR TITLE
[SYCL-MLIR]: Allow type casting from accessor to local_accessor_base

### PR DESCRIPTION
**Pull request by etiotto**

### CODE REVIEW

**Comment by whitneywhtsang** [comment]
Depends on https://github.com/intel/llvm/pull/7663, not ready for review yet.

**Comment by etiotto** [comment]
https://github.com/intel/llvm/pull/7663 landed, I am merging, and this PR will shortly be ready for review @whitneywhtsang  @victor-eds.

File: mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td [hunk 1]
```diff
@@ -149,6 +149,7 @@ def SYCL_AccessorImplDeviceType :
 def SYCL_AccessorType
     : SYCL_Type<"Accessor", "accessor",
         [SYCLInheritanceTypeTrait<"AccessorCommonType">,
+        SYCLInheritanceTypeTrait<"LocalAccessorBaseType">,
```
Comment (whitneywhtsang) [inline mlir-sycl/include/mlir/Dialect/SYCL/IR/SYCLOps.td:152]:
```suggestion
         SYCLInheritanceTypeTrait<"LocalAccessorBaseType">,
```

**Review by whitneywhtsang — APPROVED** [review]

**Comment by etiotto** [comment]
@victor-eds are you OK with the PR?

**Review by victor-eds — APPROVED** [review]

**Pull request merged**